extensions/funqy/funqy-knative-events/runtime/src/main/java/io/quarkus/funqy/runtime/bindings/knative/events/VertxRequestHandler.java

```diff
@@ -74,18 +84,57 @@ public VertxRequestHandler(Vertx vertx,
         this.currentVertxRequest = CDI.current().select(CurrentVertxRequest.class).get();
     }
 
+    private static Uni<MessageReader> messageReader(RoutingContext routingContext) {
```

Comment:
I see, you use the basic module, makes sense. Did you thought about copy-pasting the vert.x server `MessageReader` and `MessageWriter` (they should be just two simple classes) and then when quarkus supports vert.x 4, you remove them?